### ISSUE TITLE
Connect to s3 bucket through self-host docker container instance

### ISSUE THREAD

**Issue by mario16** [groupdocs-free-consulting/projects#39]
Java, MacOS, development env.

We need to convert every kind of file to pdf (groupdocs-conversion)
Another challenges: 
-generate thumbnails
-analyze content (OCR)

**Comment by tilalahmad** [comment]
@mario16 

<html>
<body>
<p style="box-sizing: inherit; margin-bottom: 1em; margin-top: 1em; overflow-wrap: break-word; line-height: 1.75; color: rgb(57, 63, 73); font-family: Roboto, system-ui, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: 0.28px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Please check out the latest release of GroupDocs.Conversion Cloud Docker Container. Now, it's possible to connect a AWS S3 cloud storage by setting environment variables:</p>

Name | Description
-- | --
S3_STORAGE_BUCKET | Bucket ID
S3_STORAGE_ACCESS_KEY | S3 API Access Key
S3_STORAGE_SECRET_KEY | S3 API Secret Key
S3_STORAGE_REGION | AWS S3 Region

<h4 style="box-sizing: inherit; margin-bottom: 1em; margin-top: 1em; overflow-wrap: break-word; color: rgb(57, 63, 73); font-family: Roboto, system-ui, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: 0.28px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Windows (PowerShell)</h4><pre style="box-sizing: inherit; font-size: 11.9px; overflow-wrap: normal; padding: 16px; overflow: auto; line-height: 1.45; background-color: rgb(249, 249, 250); border: 1px solid rgb(225, 226, 230); border-radius: 3px; color: rgb(57, 63, 73); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: 0.28px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><code class="language-powershell" style="box-sizing: inherit; font-family: &quot;Roboto Mono&quot;, consolas, monaco, monospace; font-size: 11.9px; overflow-wrap: break-word; padding: 0px; background: transparent; border-radius: 1px; margin: 0px; line-height: 1.75; word-break: normal; white-space: pre; border: 0px;">docker run `
    -p 8080:80 `
    -v "${pwd}/data:/data" `
    -e "S3_STORAGE_BUCKET=main_bucket" `
    -e "S3_STORAGE_ACCESS_KEY=XXXXXXXXXXXXXXXXXXX" `
    -e "S3_STORAGE_SECRET_KEY=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX" `
    -e "S3_STORAGE_REGION=us-west-2" `
    --name conversion_cloud `
    groupdocs/conversion-cloud
</code></pre><h4 style="box-sizing: inherit; margin-bottom: 1em; margin-top: 1em; overflow-wrap: break-word; color: rgb(57, 63, 73); font-family: Roboto, system-ui, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: 0.28px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Linux (bash)</h4><pre style="box-sizing: inherit; font-size: 11.9px; overflow-wrap: normal; padding: 16px; overflow: auto; line-height: 1.45; background-color: rgb(249, 249, 250); border: 1px solid rgb(225, 226, 230); border-radius: 3px; color: rgb(57, 63, 73); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: 0.28px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><code class="language-bash" style="box-sizing: inherit; font-family: &quot;Roboto Mono&quot;, consolas, monaco, monospace; font-size: 11.9px; overflow-wrap: break-word; padding: 0px; background: transparent; border-radius: 1px; margin: 0px; line-height: 1.75; word-break: normal; white-space: pre; border: 0px;">docker run \
    -p 8080:80 \
    -v $(pwd)/data:/data \
    -e S3_STORAGE_BUCKET=main_bucket \
    -e S3_STORAGE_ACCESS_KEY=XXXXXXXXXXXXXXXXXXX \
    -e S3_STORAGE_SECRET_KEY=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX \
    -e S3_STORAGE_REGION=us-west-2" \
    --name conversion_cloud \
    groupdocs/conversion-cloud</code></pre>
</body>
</html>

**Comment by mario16** [comment]
Hi @tilalahmad , we already started a container with the latest image 23.2 and configured the env vars.
However, when we try to list the files and folders, it ignores the "storageName" (I don't know where to set the storage name in this scenario).
The storage where always search is the bind volume.
`folderApi.getFilesList(new GetFilesListRequest("/", "bucket-name"))`

The "storageApi" always returns "true" on the "exist" function, the same occurrs on the Rest API.

_Note_: same functions are working if we use the cloud version instead of the local container.

Thanks in advance.

**Comment by mario16** [comment]
It works! 
I've noticed that _latest_ tag (of docker image) from Apr 19, 2023 at 5:01 am is older than 23.2 tag Apr 19, 2023 at 5:01 am.
With latest tag works fine.

I have some doubts, with this groupdocs server running on the container, does it check some config from cloud account like Applications (client_id, client_secret), Storages.

What if I want to track different clients? Could I configure more than one client_id on the env vars of the container?

Thanks!

**Comment by tilalahmad** [comment]
@mario16 

> It works!
> I've noticed that latest tag (of docker image) from Apr 19, 2023 at 5:01 am is older than 23.2 tag from May 4, 2023 at 5:13 am.
> With latest tag works fine.

We were publishing GroupDocs.Conversion Cloud Docker Container with the latest tag only. Recently, we found a regression issue in the latest version 23.4 and published an older version with the 23.2 tag as well. So that's why you noticed the newer date with the 23.2 release.

> I have some doubts, with this groupdocs server running on the container, does it check some config from cloud account like Applications (client_id, client_secret), Storages.

No, the Docker container does not check any information from your groupdocs cloud hosted account. The Docker container uses a metered license and it periodically sends the API usage to our billing server.

>What if I want to track different clients? Could I configure more than one client_id on the env vars of the container?
I will check this out and get back to you soon.

**Comment by mario16** [comment]
Thank you very much. 
I will wait for the other response.

**Comment by tilalahmad** [comment]
@mario16 

>What if I want to track different clients? Could I configure more than one client_id on the env vars of the container?

I am afraid that currently, we only support a single Client ID in the Docker container. However, we have logged a ticket to configure multiple Client IDs and will notify you as soon as the ticket is resolved.

**Comment by mario16** [comment]
Awesome! I will stay tune, we are going to purchase the license.

**Comment by mario16** [comment]
What happen if we need to use different s3 buckets?
Can I set a s3 bucket on "storageName" attribute of the ConvertSettings? Or is the one on the env vars the only allowed?

**Comment by tilalahmad** [comment]
@mario16 

> What happen if we need to use different s3 buckets?
> Can I set a s3 bucket on "storageName" attribute of the ConvertSettings? Or is the one on the env vars the only allowed?

Yes, you are right. You can only have one bucket configured at the time of container initialization.
